Documentation/guides/drivers.rst

```diff
@@ -0,0 +1,211 @@
+        200aaa00: 5752 5445 5354 3520 5458 5420 1800 0000  WRTEST5 TXT ....
+        200aaa10: 0000 0000 0000 0000 0000 5800 1500 0000  ..........X.....
+        200aaa20: e552 5445 5854 3620 5458 5420 1800 0000  .RTEXT6 TXT ....
+        200aaa30: 0000 0000 0000 0000 0000 5600 1200 0000  ..........V.....
+        200aaa40: 5752 5445 5354 3620 5458 5420 1800 0000  WRTEST6 TXT ....
+        200aaa50: 0000 0000 0000 0000 0000 5600 1200 0000  ..........V.....
+        200aaa60: 0000 0000 0000 0000 0000 0000 0000 0000  ................
+        200aaa70: 0000 0000 0000 0000 0000 0000 0000 0000  ................
+        200aaa80: 0000 0000 0000 0000 0000 0000 0000 0000  ................
+        200aaa90: 0000 0000 0000 0000 0000 0000 0000 0000  ................
+        200aaaa0: 0000 0000 0000 0000                      ........
+
+
+NuttX Drivers as a Reference
+----------------------------
+
+If you're not porting a NuttX driver from another architecture, it still helps to look at other similar NuttX
+drivers, if there are any. For instance, when implementing an Ethernet driver, look at other NuttX Ethernet drivers;
+for an SD Card driver, look at other NuttX Ethernet drivers. Even if the chip-specific code won't be the same, the
+structure to interface with NuttX can be used.
+
+Using Chip Datasheets
+---------------------
+
+To port or write a driver, you'll have to be familiar with the information in the chip datasheet. Definitely find
+the datasheet for your chip, and read the sections relevant to the peripheral you're working with. Doing so ahead
+of time will save a lot of time later.
+
+Another thing that's often helpful is to refer to sample code provided by the manufacturer, or driver code from
+another operating system (like U-Boot, Zephyr, or FreeBSD) while referring to the datasheet — seeing how working
+code implements the necessary algorithms often helps one understand how the driver needs to work.
+
+* How to use a datasheet
+
+  Key pieces of information in System-on-a-Chip (SoC) datasheets are usually:
+
+  * Chip Architecture Diagram — shows how the subsections of the chip (CPU, system bus, peripherals, I/O, etc.) connect
+    to each other.
+  * Memory Map — showing the location of peripheral registers in memory. This info usually goes into a header file.
+  * DMA Engine — if Direct Memory Access (DMA) is used, this may have info on how to use it.
+  * Peripheral — the datasheet usually has a section on how the peripheral works. Key parts of this include:
+
+    * Registers List — name and offset from the base memory address of the peripheral. This needs to go into a header
+      file.
+    * Register Map — what is the size of each register, and what do the bits mean? You will need to create ``#defines``
+      in a header file that your code will use to operate on the registers. Refer to other driver header files for
+      examples.
+
+`Logic analyzers <https://en.wikipedia.org/wiki/Logic_analyzer>`_
```

Comment:
Do we want this whole heading to be a link?  It's a little jarring.

Reply:
@btashton I moved the link into the paragraph, and left the header as plain text.